react/components/HighlightOverlay/index.test.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react'
+import { act, fireEvent, render } from '@vtex/test-tools/react'
+import { State } from './typings'
+import HighlightOverlay from './index'
+
+const extensionPointName = 'welcome'
+
+function createMockDiv(width: number, height: number) {
+  const renderProviderDiv = document.createElement('div')
+  renderProviderDiv.setAttribute('class', 'render-provider')
+  Object.assign(renderProviderDiv.style, {
+    width: `1440px`,
+    height: `900px`,
+  })
+
+  const div = document.createElement('div')
+  Object.assign(div.style, {
+    width: `${width}px`,
+    height: `${height}px`,
+  })
+
+  // we have to mock this for jsdom.
+  div.getBoundingClientRect = () => ({
+    width,
+    height,
+    top: 0,
+    left: 0,
+    right: width,
+    bottom: height,
+    x: width,
+    y: height,
+  })
+  div.textContent = 'Welcome'
+  div.setAttribute('data-extension-point', extensionPointName)
+
+  renderProviderDiv.getBoundingClientRect = () => ({
+    width,
+    height,
+    top: 0,
+    left: 0,
+    right: width,
+    bottom: height,
+    x: width,
+    y: height,
+  })
+  renderProviderDiv.appendChild(div)
+  return renderProviderDiv
+}
+
+const mockBlocks: State['sidebarBlocksMap'] = {
+  welcome: {
+    isEditable: true,
+    title: 'Welcome Banner',
+  },
+}
+
+function setUpDom() {
+  Object.defineProperty(HTMLElement.prototype, 'scrollWidth', {
+    configurable: true,
+    get: function getScrollWidth() {
+      return this._scrollWidth || 0
+    },
+    set(val) {
+      this._scrollWidth = val
+    },
+  })
+  // @ts-ignore
```

Comment:
:eyes:

Reply:
The next line (`document.body.scrollWidth = 1440`) gives the error `Cannot assign to 'scrollWidth' because it is a read-only property.` which is true on the browser. The error is indeed an error as the `scrollWidth` doesn't when reassigned (so this could be a bug 🐛 if you're relying on this behaviour).

However, when using `jsdom` there's no other way to change this value, and since this property is used in our hooks, we need to set to a value other than the default `0`.